### PR TITLE
encryption: remove key secret finalizer before pruning

### PR DESCRIPTION
**Pull request by sttts**
The key secret have a finalizer. This must be removed before deletion.

### CODE REVIEW

**Comment by sttts** [comment]
/hold 

until manually tested (we don't have e2e tests for this).

File: pkg/operator/prunecontroller/prune_controller.go [hunk 1]
```diff
@@ -96,7 +100,32 @@ func (c *PruneController) sync() error {
...
+				}
+				newFinalizers = append(newFinalizers, f)
+			}
+			s.Finalizers = newFinalizers
+
+			_, err = c.secretGetter.Secrets(s.Namespace).Update(s)
+			return err
+		})
+
+		if err := c.secretGetter.Secrets(s.Namespace).Delete(s.Name, nil); err != nil && !errors.IsNotFound(err) {
```
Comment (p0lyn0mial) [inline pkg/operator/prunecontroller/prune_controller.go]: ok, so regardless of the status of the previous operation (success, failure) remove the secret, why not :)

File: pkg/operator/prunecontroller/prune_controller.go [hunk 2]
```diff
@@ -96,7 +100,32 @@ func (c *PruneController) sync() error {
...
+			newFinalizers := make([]string, 0, len(s.Finalizers))
+			for _, f := range s.Finalizers {
+				if f == encryptionsecrets.EncryptionSecretFinalizer {
+					continue
+				}
+				newFinalizers = append(newFinalizers, f)
+			}
+			s.Finalizers = newFinalizers
+
+			_, err = c.secretGetter.Secrets(s.Namespace).Update(s)
```
Comment (p0lyn0mial) [inline pkg/operator/prunecontroller/prune_controller.go]: if `len(newFinalizers) == len(s.Finalizers)` do not update, wdyt ?

Comment (sttts) [reply pkg/operator/prunecontroller/prune_controller.go]: done

File: pkg/operator/prunecontroller/prune_controller.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/prunecontroller/prune_controller.go]: Worth a try.

**Pull request merged**